internal/report/store_memory.go

```diff
@@ -0,0 +1,172 @@
+package report
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/kubewarden/audit-scanner/internal/constants"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+// MemoryPolicyReportStore is an implementation of `PolicyReportStore`
+// that uses in-memory cache to store `PolicyReports` and `ClusterPolicyReports`
+type MemoryPolicyReportStore struct {
+	// prCache is a map between a namespace and a PolicyReport
+	prCache map[string]PolicyReport
+
+	// cprCache is a map between a name and a ClusterPolicyReport
+	cprCache map[string]ClusterPolicyReport
+}
+
+func NewMemoryPolicyReportStore() (*MemoryPolicyReportStore, error) {
```

Comment:
Since it never fails, we could omit `error` from the return type

Reply:
I had mixed feelings about this one, I was thinking of keeping all constructors for all `PolicyReportStore` implementations coherent, but I'm now thinking that doesn't make sense because we'll have different dependencies for each store. I'll remove the `error`